src/SourceLocator/Ast/Strategy/NodeToReflection.php

```diff
@@ -28,6 +29,10 @@ public function __invoke(
     ): ReflectionClass|ReflectionConstant|ReflectionFunction {
         $namespaceName = $namespace?->name?->name;
 
+        if ($namespaceName === '') {
+            throw new LogicException('Namespace name should never be empty');
+        }
```

Comment:
Is this something we can test at all? Or is it an upstream type refinement required?

Reply:
the upstream type is already refined, but the change itself is not yet released

https://github.com/nikic/PHP-Parser/pull/993

Reply:
Let's then roll with this as-is: a lock file update will expose the type contradiction later on 👍